llvm/lib/Object/ELFObjectFile.cpp

```diff
@@ -517,6 +519,73 @@ StringRef ELFObjectFileBase::getAMDGPUCPUName() const {
   }
 }
 
+StringRef ELFObjectFileBase::getNVPTXCPUName() const {
+  assert(getEMachine() == ELF::EM_CUDA);
+  unsigned SM = getPlatformFlags() & ELF::EF_CUDA_SM;
```

Comment:
If SM is only used once, just inline it into the use site

Reply:
I mostly copied this from the existing AMDGPU version and kept the style.

Reply:
🤷 my feeling is two wrongs don't make a right.  Especially since AMD *does* have CPUs.  But nbd.